code-challenges.test.js

```diff
@@ -45,23 +82,43 @@ const studyMinutesWeek2 = {
   wednesday: 60,
   thursday: 20,
   friday: 15,
-  saturday: 65
-}
+  saturday: 65,
+};
 // Expected output: [10, 15, 20, 45, 60, 65, 100]
-
+const testArray = (smw) => {
+  return Object.values(smw).sort();
+};
+// so the test passes with the first parameter but not the second one. i'm not sure why. moving to the next question.
 // b) Create the function that makes the test pass.
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+//sounds like the fibinnacci function from above. except the array is already given to us and we are applying this accumulating function over the arrays length.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const accountTransactions1 = [100, -17, -23, -9]
+describe("accuArray", () => {
+  it("returns an array of the accumulating sum", () => {
+    expect(accuArray(accountTransactions1)).toEqual([100, -17, -23, -9]);
+    //expect(accuArray(accountTransactions2)).toEqual([250, -89, 100, -96]);
+    expect(accuArray(accountTransactions3)).toEqual([]);
+  });
+});
+//test failed successfully. moving on to create the function.
+
+const accountTransactions1 = [100, -17, -23, -9];
 // Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
+const accountTransactions2 = [250, -89, 100, -96];
 // Expected output: [250, 161, 261, 165]
 
-const accountTransactions3 = []
+const accountTransactions3 = [];
 // Expected output: []
-
+//i'm not sure how to have the function add the current index with the index before it. i feel that will be crucial to completing this function
+const accuArray = (array) => {
+  let newArr = [];
+  for (let i = 1; i < array.length; i++) {
+    const element = array[i] + array[i--];
+  }
+  return newArr;
+};
+// im not sure if just me, but jest doesn't seem to work. it was working fine i enter yarn jest and nothing happens. it says no license field and i restarted my terminal ran yarn add jest and same thing when i enter yarn jest. i think my function is really close to the answer but not sure how to check or move forward. i really hope to get better with jest and able to finish thes kind of challenges better next time.
```

Comment:
Please reach out on slack if you are running into a blocker like this.  Not being able to see the output of your tests is going to be very limiting and it is important to work through all kinds of errors.  I would also really like to see more pseudocode breaking down your process.  I appreciate that you are providing some comments, however I also want to see the pseudocode on the logic provided in your code.  For instance, you created a for loop, but your pseudocode does not mention the need for iteration.